backend/src/main/java/ca/bc/gov/hlth/mohums/controller/UsersController.java

```diff
@@ -67,116 +65,72 @@ public UsersController(WebClientService webClientService, @Value("${config.vanit
 
-            // filter user list;
-            logger.debug("numbers of users before filtering by last Log Date: " + users.size());
-            
             List<Object> filteredUsersByLastLog = new ArrayList<>();
             for (Object user : users) {
                 String userId = ((LinkedHashMap) user).get("id").toString();
-                if(!userId.isEmpty()){
-                    
-                    // Filter only the events for the current user 
-                    // TODO better way to do this will be to build a map of Event per users
-                    Predicate<Object> byUserId = (event -> {
-                        Object userIdFromEvent = ((LinkedHashMap) event).get("userId");
-                        return userIdFromEvent!= null && userIdFromEvent.equals(userId);
-                            });
-   
-                    var eventsByUserID = allEventsLastLog.stream().filter(byUserId)
-                                    .collect(Collectors.toList());
-                
-                    logger.debug("number of events By UserID: " + eventsByUserID.size());
-                    
-                    // sort list of events by time, descending
-                    Comparator compareEventTime = (Comparator<Object>) (Object event1, Object event2) -> {
-                        Long time1 = (Long)(((LinkedHashMap) event1).get("time"));
-                        Long time2 = (Long)(((LinkedHashMap) event2).get("time"));
-                        Instant a = Instant.ofEpochMilli(time1);
-                        Instant b = Instant.ofEpochMilli(time2);
-                        return b.compareTo(a);
-                    };
-                    Collections.sort(eventsByUserID, compareEventTime);
-                  
-                    // Verify if Last Log date match the criteria
-                    // add lastLogDate to user. Date will be formatted in Frontend
-                    if (!eventsByUserID.isEmpty()){
+                if (!userId.isEmpty()){
+
+                    if (!CollectionUtils.isEmpty(loginEventsByUser.get(userId))) {
                         LocalDate lastLogAfterDate, lastLogBeforeDate;
-                        
-                        Object userLastLogTime = ((LinkedHashMap)eventsByUserID.get(0)).get("time");
+
+                        Object userLastLogTime = loginEventsByUser.get(userId).stream()
+                                .max(Comparator.comparing(o -> (Long) ((LinkedHashMap) o).get("time")))
+                                .map(o -> ((LinkedHashMap) o).get("time")).get();
+
                         LocalDate userLastLogLocalDate = LocalDate.ofInstant(Instant.ofEpochMilli((Long) userLastLogTime),  ZoneId.of("America/Los_Angeles"));
-                        
+                        //Or we could send the time as is to the Frontend, without formatting
+                        ((LinkedHashMap) user).put("lastLogDate", userLastLogLocalDate.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")));
```

Comment:
I think we should just send it to the Frontend without formatting. That's what the `Events` API behaves.